iTunesUDataPurge/RemoveITunesUCourses.java

```diff
@@ -478,5 +501,186 @@ public static void fileUrl(String fAddress, String localFileName, String destina
 			}
 		}
 	}
+
+	public static void deletePage(String searchServer, String siteId, String username, String password) throws Exception{
+
+		String pageToDelete = "iTunes U";
+
+		// Create SOAP Connection
+		SOAPConnectionFactory soapConnectionFactory = SOAPConnectionFactory.newInstance();
+		SOAPConnection soapConnection = soapConnectionFactory.createConnection();
+
+		// Send SOAP Message to SOAP Server
+		String sakaiLoginUrl = searchServer + "sakai-axis/SakaiLogin.jws?wsdl";
+		String sakaiScriptUrl = searchServer + "sakai-axis/SakaiScript.jws?wsdl";
+
+		//Making the createLoginRequest call. Returns a response message containing the session id that should be active on server.
+		SOAPMessage soapResponse = soapConnection.call(createLoginRequest(searchServer, username, password), sakaiLoginUrl);
+
+		// print SOAP Response
+		System.out.print("Response SOAP Message:");
+		soapResponse.writeTo(System.out);
+
+		//Parses Soap Response to get session id as string
+		String sessionId = getSessionId(soapResponse);
+		System.out.println("");
+		System.out.println("Session Id:" + sessionId);
+
+		//Test validity of session id by calling Check Session function
+		soapResponse = soapConnection.call(createCheckSessionRequest(searchServer, sessionId), sakaiScriptUrl);
+		System.out.print("Response SOAP Message:");
+		soapResponse.writeTo(System.out);
+
+		String checkedSessionId = getSessionId(soapResponse);
+		System.out.println("");
+		System.out.println("Checked Session Id: " + checkedSessionId);
+
+		//while there are still sites to delete:
+		if(!checkedSessionId.equals("null")){
+			
+			// if session id is not null then remove site
+			soapResponse = soapConnection.call(createRemovePageRequest(searchServer, sessionId, siteId, pageToDelete), sakaiScriptUrl);
+			System.out.print("Response SOAP Message:");
+			soapResponse.writeTo(System.out);
+			System.out.println("");
+			
+		}
+		else{
+			//else get new session id and then process site
+			soapResponse = soapConnection.call(createLoginRequest(searchServer, username, password), sakaiLoginUrl);
```

Comment:
not sure why we need the create another login session here if the previous one failed. There is no subsequent action after this if the session is created (as second try)